platform/pulpcore/app/viewsets/content.py

```diff
@@ -44,6 +45,28 @@ class ContentViewSet(NamedModelViewSet):
     serializer_class = ContentSerializer
     filter_class = ContentFilter
 
+    @transaction.atomic
+    def create(self, request):
```

Comment:
I haven't seen this kind of _helper_ method on other models.  So far, we've made a deliberate effort to keep the Pulp domain model and the data model separated.  It seems like this functionality belongs somewhere else.